capa/features/extractors/ghidra/helpers.py

```diff
@@ -0,0 +1,23 @@
+# Copyright (C) 2020 Mandiant, Inc. All Rights Reserved.
+# Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at: [package root]/LICENSE.txt
+# Unless required by applicable law or agreed to in writing, software distributed under the License
+#  is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and limitations under the License.
+from typing import Any, Dict, Tuple, Iterator, Optional
+
+
+def find_byte_sequence(seq: bytes) -> Iterator[int]:
+    """yield all ea of a given byte sequence
+
+    args:
+        start: min virtual address
+        end: max virtual address
+        seq: bytes to search e.g. b"\x01\x03"
+    """
+    seqstr = " ".join([f"{b:02x}" for b in seq])
+    print(seqstr)
+    yield currentAddress
```

Comment:
(how) does this work (yet)?

Reply:
Please see d043647024ef739e4274fbc4b2979de370ae0f52. The Ghidra API has made it (somewhat) convenient to parse for all of the xor'd bytes in the file. But there are a couple of issues which I'll address here:

 For the statically xor’d bytes that we check against, I would have to re-encode them back into UTF-8 in order for the Ghidra api’s function (findBytes()) to parse the file. However the advantage with parsing using findBytes() is that it handles any invalid memory accesses and iterates over all of the segments that need to be parsed. Another challenge comes from a string handling issue with Java as there are special characters generated that it wants us to have to escape. As a temporary workaround, pending a more solid solution, I have the helper function throw an exception. See the picture below. As a note, this is the only variation from all of the xor’d bytes that needs to have the exception handled.

![image](https://github.com/mandiant/capa/assets/66766340/74ae6a80-25e8-4296-ae72-372c3e18c9ea)
> Java would like us to escape the '[' character ('\\[')

Bytes that are not UTF-8 encodable are ignored by the `decode()` function and the string still gets passed to `findBytes()`.

Reply:
Totally understand the problem you're having here. We actually had a similar issue with IDA that we resolved using a wrapper that converts the byte sequence to a format suitable for the IDA's `find_binary` API: https://github.com/mandiant/capa/blob/33a3170bc4160dbfe925ef053e61823ece245996/capa/features/extractors/ida/helpers.py#L20-L35

We can do something similar for Ghidra e.g. we want to search for the byte sequence `0x7f`, `0x45` we can call Ghidra's `findBytes` as follows:

```python
>>> hex(findBytes(currentProgram.getMinAddress(), "\x7f\x45").getOffset())
'0x100000'
```

and can write a wrapper that converts `0x7f`, `0x45` into the string `\x7f\x45` and pass that to Ghidra's `findBytes` API. Thoughts?

Reply:
The `\\` eliminates the error you described above:

```python
>>> findBytes(currentProgram.getMinAddress(), "\\x4c\\x5b")
>>> 
```

Reply:
This implementation worked much better! Please see: f36f48bdba2eeb73a95423b62c22053cee45bc98